dp-identity-data-migration/groupsdataextraction/group_extraction.go

```diff
@@ -0,0 +1,283 @@
+		fmt.Println(user)
+	}
+	return userList, nil
+}
+
+func convert_to_slice_group(input group) []string {
+	return []string{
+		input.GroupName,
+		input.UserPoolId,
+		input.Description,
+		input.RoleArn,
+		input.Precedence,
+		input.LastModifiedDate,
+		input.CreationDate,
+	}
+}
+
+func convert_to_slice_group_user(input user_group_csv) []string {
+	return []string{
+		input.Username,
+		input.Groups,
+	}
+}
+
+func readCsvFile(filePath string) int {
+	f, err := os.Open(filePath)
+	if err != nil {
+		fmt.Println("Unable to read input file "+filePath, err)
+	}
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		fmt.Println("Unable to parse file as CSV for "+filePath, err)
+	}
+
+	return len(records)
+}
+
+func getClient(conf config) zebedee.Client {
+	httpCli := zebedee.NewHttpClient(time.Second * 5)
+	return zebedee.NewClient(conf.host, httpCli)
+}
+
+var emptylist []string
+
+func main() {
+
+	conf := readConfig()
+	zebCli := getClient(conf)
+
+	c := zebedee.Credentials{
+		Email:    conf.user,
+		Password: conf.pword,
+	}
+
+	sess, err := zebCli.OpenSession(c)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	groupList, err := getgroups(zebCli, sess)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	groups_csvfile, err := os.Create(conf.groups_filename)
+	if err != nil {
+		fmt.Printf("failed creating file: %s", err)
+		os.Exit(1)
+	}
+	csvwriter := csv.NewWriter(groups_csvfile)
+	csvwriter.Write(convert_to_slice_group(group_header))
+	for _, zebedeegroup := range groupList.Teams {
+		tmp := group{
+			GroupName:        zebedeegroup.Name,
+			UserPoolId:       "",
+			Description:      zebedeegroup.Name,
+			RoleArn:          "",
+			Precedence:       "10",
+			LastModifiedDate: "",
+			CreationDate:     "",
+		}
+		csvwriter.Write(convert_to_slice_group(tmp))
+
+	}
+	csvwriter.Flush()
+	groups_csvfile.Close()
+	fmt.Println("========= ", conf.groups_filename, "file validiation =============")
+	actualrowcount := readCsvFile(conf.groups_filename) - 1
+	if actualrowcount != len(groupList.Teams) || csvwriter.Error() != nil {
+		fmt.Println("There has been an error... ")
+	}
+
+	fmt.Println("Expected row count: - ", len(groupList.Teams))
+	fmt.Println("Actual row count: - ", actualrowcount)
+	fmt.Println("csv Errors ", csvwriter.Error())
+	fmt.Println("=========")
+
+	// tmplen := 0
+	tmpusergroups := make(map[string][]string)
+
+	usergroups_csvfile, err := os.Create(conf.groupusers_filename)
+	if err != nil {
+		fmt.Printf("failed creating file: %s", err)
+		os.Exit(1)
+	}
+	csvwriter = csv.NewWriter(usergroups_csvfile)
+	csvwriter.Write(convert_to_slice_group_user(header_user_groups))
+
+	userList, err := getUsers(zebCli, sess)
+	if err != nil {
+		fmt.Println("Theres been an issue", err)
+		os.Exit(1)
+	}
+
+	for _, user := range userList {
+
+		_, isKeyPresent := tmpusergroups[user.Email]
+		if !isKeyPresent {
+			tmpusergroups[user.Email] = emptylist
```

Comment:
Rather than do this, I'd do something like:
```suggestion
			tmpusergroups[user.Email] = emptylist
			tmpusergroups[user.Email] = make([]string,0)
```